src/components/Colorpicker.vue

```diff
@@ -0,0 +1,179 @@
+<!--
+Nextcloud - Tasks
+
+@author Raimund Schlüßler
+@author Raghu Nayyar
+@author Georg Ehrke
+@author John Molakvoæ
+@copyright 2018 Raimund Schlüßler <raimund.schluessler@mailbox.org>
+@copyright 2018 Raghu Nayyar <beingminimal@gmail.com>
+@copyright 2018 Georg Ehrke <oc.list@georgehrke.com>
+@copyright 2018 John Molakvoæ <skjnldsv@protonmail.com>
+
+This library is free software; you can redistribute it and/or
+modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+License as published by the Free Software Foundation; either
+version 3 of the License, or any later version.
+
+This library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+
+You should have received a copy of the GNU Affero General Public
+License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+
+-->
+
+<template>
+	<div class="colorpicker">
+		<ul class="colorpicker-list">
+			<li v-for="color in colors"
+				:key="color"
+				:class="{ selected: (color===selectedColor) }"
+				:style="{'background-color': color}"
+				@click="pick(color)"
+			/>
+			<li v-if="!supportsColorPicker"
+				:style="{'background-color': random}"
+				class="randomcolour"
+				@click="randomizeColour"
+			>
+				<span class="icon icon-random" />
+			</li>
+			<label v-if="supportsColorPicker"
+				:style="{'background-color': selectedColor}"
+				class="color-selector-label"
+			>
+				<input :value="selectedColor"
+					type="color"
+					class="color-selector"
+					@change="pick($event.target.value)"
+				>
+			</label>
+		</ul>
+	</div>
+</template>
+
+<script>
+export default {
+	name: 'Colorpicker',
+	props: {
+		selectedColor: {
+			type: String,
+			default: '#31CC7C'
+		}
+	},
+	data() {
+		return {
+			random: '#31CC7C',
+			colors: [
+				'#31CC7C',
+				'#317CCC',
+				'#FF7A66',
+				'#F1DB50',
+				'#7C31CC',
+				'#CC317C',
+				'#3A3B3D',
+				'#CACBCD'
+			]
+		}
+	},
+	methods: {
+		supportsColorPicker: function() {
+			const inputElement = document.createElement('input')
+			inputElement.setAttribute('type', 'color')
+			return inputElement.type === 'color'
+		},
+		randomizeColour: function() {
+			this.random = this.randColour()
+			this.pick(this.random)
+		},
+		pick: function(color) {
+			this.$emit('color-selected', color)
+		},
+		/*
+		* Generate a random colour with the core generator
+		*/
+		randColour: function() {
+			if (typeof String.prototype.toHsl === 'function') {
```

Comment:
We change this a while ago :)
https://github.com/nextcloud/server/blob/master/core/js/placeholder.js#L64